addons/binding/org.openhab.binding.avmfritz/src/main/java/org/openhab/binding/avmfritz/handler/DeviceHandler.java

```diff
@@ -147,15 +147,8 @@ private synchronized void onUpdate() {
     @Override
     public void handleCommand(ChannelUID channelUID, Command command) {
         logger.debug("Handle command {} for channel {}", channelUID.getIdWithoutGroup(), command);
-        FritzahaWebInterface fritzBox = null;
-        if (!getThing().getThingTypeUID().equals(PL546E_STANDALONE_THING_TYPE)) {
-            Bridge bridge = getBridge();
-            if (bridge != null && bridge.getHandler() instanceof BoxHandler) {
-                fritzBox = ((BoxHandler) bridge.getHandler()).getWebInterface();
-            }
-        } else {
-            fritzBox = getWebInterface();
-        }
+        final FritzahaWebInterface fritzBox = PL546E_STANDALONE_THING_TYPE.equals(getThing().getThingTypeUID())
+                ? getWebInterface() : ((BoxHandler) getBridge().getHandler()).getWebInterface();
```

Comment:
@martinvw We need a lot more checks here. I don't like this lines of code. I moved the complete logic to the `getWebInterface()` method.